drivers/sensor/lis2dw12/lis2dw12.h

```diff
@@ -0,0 +1,220 @@
+	defined(CONFIG_LIS2DW12_ODR_RUNTIME)
+	#define LIS2DW12_DEFAULT_ODR	LIS2DW12_ODR_100HZ_VAL
+#elif defined(CONFIG_LIS2DW12_ODR_200)
+	#define LIS2DW12_DEFAULT_ODR	LIS2DW12_ODR_200HZ_VAL
+#elif defined(CONFIG_LIS2DW12_ODR_400)
+	#define LIS2DW12_DEFAULT_ODR	LIS2DW12_ODR_400HZ_VAL
+#elif defined(CONFIG_LIS2DW12_ODR_800)
+	#define LIS2DW12_DEFAULT_ODR	LIS2DW12_ODR_800HZ_VAL
+#elif defined(CONFIG_LIS2DW12_ODR_1600)
+	#define LIS2DW12_DEFAULT_ODR	LIS2DW12_ODR_1600HZ_VAL
+#endif
+
+/* Return ODR reg value based on data rate set */
+#define LIS2DW12_ODR_TO_REG(_odr) \
+	((_odr <= 1) ? LIS2DW12_ODR_1_6HZ_VAL : \
+	 (_odr <= 12) ? LIS2DW12_ODR_12_5HZ_VAL : \
+	 ((31 - __builtin_clz(_odr / 25))) + 3)
+
+#define LIS2DW12_CTRL2_ADDR		0x21
+#define LIS2DW12_BDU_MASK		BIT(3)
+#define LIS2DW12_RESET_MASK		BIT(6)
+#define LIS2DW12_BOOT_MASK		BIT(7)
+#define LIS2DW12_CTRL3_ADDR		0x22
+#define LIS2DW12_LIR_MASK		BIT(4)
+
+#define LIS2DW12_CTRL4_ADDR		0x23
+#define LIS2DW12_INT1_DRDY		BIT(0)
+
+#define LIS2DW12_CTRL5_ADDR		0x24
+#define LIS2DW12_INT2_DRDY		BIT(0)
+
+#define LIS2DW12_CTRL6_ADDR		0x25
+#define LIS2DW12_FS_MASK		0x30
+
+enum lis2dh_fs {
+	LIS2DW12_FS_2G_VAL,
+	LIS2DW12_FS_4G_VAL,
+	LIS2DW12_FS_8G_VAL,
+	LIS2DW12_FS_16G_VAL
+};
+
+/* FS reg value from Full Scale */
+#define LIS2DW12_FS_TO_REG(_fs)	(30 - __builtin_clz(_fs))
+
+#if defined(CONFIG_LIS2DW12_ACCEL_RANGE_RUNTIME) || \
+	defined(CONFIG_LIS2DW12_ACCEL_RANGE_2G)
+	#define LIS2DW12_ACC_FS		LIS2DW12_FS_2G_VAL
+#elif defined(CONFIG_LIS2DW12_ACCEL_RANGE_4G)
+	#define LIS2DW12_ACC_FS		LIS2DW12_FS_4G_VAL
+#elif defined(CONFIG_LIS2DW12_ACCEL_RANGE_8G)
+	#define LIS2DW12_ACC_FS		LIS2DW12_FS_8G_VAL
+#elif defined(CONFIG_LIS2DW12_ACCEL_RANGE_16G)
+	#define LIS2DW12_ACC_FS		LIS2DW12_FS_16G_VAL
+#endif
+
+#define LIS2DW12_OUT_T_REG		0x26
+
+#define LIS2DW12_STATUS_REG		0x27
+#define LIS2DW12_STS_XLDA_UP		0x01
+
+#define LIS2DW12_OUT_X_L_ADDR		0x28
+
+/* Acc Gain value in ug/LSB in High Perf mode */
+#define LIS2DW12_FS_2G_GAIN		244
+#define LIS2DW12_FS_4G_GAIN		488
+#define LIS2DW12_FS_8G_GAIN		976
+#define LIS2DW12_FS_16G_GAIN		1952
+
+#define LIS2DW12_SHFT_GAIN_NOLP1	2
+#define LIS2DW12_ACCEL_GAIN_DEFAULT_VAL LIS2DW12_FS_2G_GAIN
+#define LIS2DW12_FS_TO_GAIN(_fs, _lp1) \
+		(LIS2DW12_FS_2G_GAIN << ((_fs) + (_lp1)))
+
+/* shift value for power mode */
+#define LIS2DW12_SHIFT_PM1		4
+#define LIS2DW12_SHIFT_PMOTHER		2
+
+/**
+ * struct lis2dw12_device_config - lis2dw12 hw configuration
+ * @bus_name: Pointer to bus master identifier.
+ * @pm: Power mode (lis2dh_powermode).
+ * @int_gpio_port: Pointer to GPIO PORT identifier.
+ * @int_gpio_pin: GPIO pin number connecter to sensor int pin.
+ * @int_pin: Sensor int pin (int1/int2).
+ */
+struct lis2dw12_device_config {
+	const char *bus_name;
+	enum lis2dh_powermode pm;
+#ifdef CONFIG_LIS2DW12_TRIGGER
+	const char *int_gpio_port;
+	u8_t int_gpio_pin;
+	u8_t int_pin;
+#endif /* CONFIG_LIS2DW12_TRIGGER */
+};
+
+struct lis2dw12_data;
```

Comment:
Why is this here?

Reply:
Because lis2dw12_data and lis2dw12_tf structures are referrring one to each other. So, in order to compile properly, I need to declare lis2dw12_data before lis2dw12_tf, and then define all the
fields properly after. It is a common practice done also in other sensor drivers too
(LSM6DSL and LIS2DS12)

Reply:
It wasn't initially obvious that this was a forward declaration. It's helpful to comment as such.

Reply:
OK, I'm adding a comment.